### PR TITLE
Separate shallow and deep color maps for some analysis tasks

### LINKED ISSUE #642: Consider separate range for surface and depth in depth-slice climatology maps
Color map ranges that are appropriate for the surface are often bad at depth and visa versa, see #640.  It would be useful to define a "surface" range and a "deep" range, together with a depth at which they switch.  "top" would always use the "surface" range (though it can have any depth below ice shelves) and "bot" would always the "deep" range (though it can become very shallow).

### PR DESCRIPTION
**Pull request by xylar**
In the SOSE and WOA18 climatology maps and only for temperature, salinity and potential density,
not velocities, set up a different color map (with a different range) for shallow depths (above -500 m by default) and deeper depths.

closes #642

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

See this run for examples of this change:
https://portal.nersc.gov/project/m3412/mpas_analysis_output/separate_shallow_deep_colormap/20191018.A_WCYCL1850-DIB-ISMF_CMIP6.ne30_oECv3wLI.cori-knl.testNewGM/yrs1-10/ocean/index.html

See this example for the previous way:
https://portal.nersc.gov/project/m3412/mpas_analysis_output/20191018.A_WCYCL1850-DIB-ISMF_CMIP6.ne30_oECv3wLI.cori-knl.testNewGM/yrs1-10/ocean/index.html#temperatureSose

**Comment by xylar** [comment]
@milenaveneziani (and anyone else), I definitely see some up sides and down sides to this approach.
### sea surface
new:
image
old (not the same run, just so it's clear):
image
Clearly the new is preferable.

### sea floor
new:
image
old:
image

The transition between -400 and -600 is kind of abrupt, though.  Does it seem bad enough that users will get the wrong idea?

File: mpas_analysis/config.default [hunk 1]
```diff
@@ -1417,9 +1421,36 @@ colormapTypeResult = continuous
 # the type of norm used in the colormap
 normTypeResult = linear
 # A dictionary with keywords for the norm
-normArgsResult = {'vmin': -2., 'vmax': 2.}
+normArgsResult = {'vmin': -3., 'vmax': 10.}
 # place the ticks automatically by default
-# colorbarTicksResult = numpy.linspace(-2., 2., 9)
+# colorbarTicksResult = numpy.linspace(-3., 10., 9)
```
Comment (milenaveneziani) [inline mpas_analysis/config.default]: maybe down to -2 is enough?

**Comment by milenaveneziani** [comment]
> The transition between -400 and -600 is kind of abrupt, though. Does it seem bad enough that users will get the wrong idea?

maybe we can push the transition depth down to -1000?

Also, I tend to like the bottom temperature plots better with [-2, 2]: what do people think?

**Comment by milenaveneziani** [comment]
This looks great. Thanks for adding these options @xylar.

File: mpas_analysis/config.default (hunk 1 above)
Comment (xylar) [reply mpas_analysis/config.default]: This was to be consistent with the deeper values, which I want to be symmetric around 0 (separating "cold" from "warm" shelf regions) and which I expanded to -3 to 3.  Also, temperatures under ice shelves get down to -2.5 or below.

**Comment by xylar** [comment]
> maybe we can push the transition depth down to -1000?

That's an option.

**Comment by xylar** [comment]
I went back to the previous range (-2 to 2 at the bottom and -2 to 10)  Here's what that looks like for the examples above:
### sea surface
new:
image
old:
image

### sea floor
new (should now be identical to the old):
image
old:
image

**Comment by xylar** [comment]
Various example config file in `configs` still contain the old `[climatologyMapWoaTemperature]` and `[climatologyMapWoaSalinity]` sections.  However, I plan to remove those as part of #640.

**Pull request merged**